client/my-sites/comments/comment-navigation/index.jsx

```diff
@@ -249,11 +249,16 @@ const mapStateToProps = ( state, { commentsPage, siteId } ) => {
 			};
 		}
 	} );
+	const selectedComments = getSelectedComments( state, siteId );
 
 	return {
+		isSelectedAll:
+			difference( commentsPage, map( selectedComments, ( { commentId } ) => commentId ) ).length ===
```

Comment:
I assume this is in case some or all `selectedComments` are not contained in the current page.
Is it something that can actually happen _right now_, or is it future-proofing in case we add multi-page selection?

Reply:
You hit the jackpot. Having external state makes it really easy to have selections across pages. The original code just checked the `length` of the selected comments to be the same as the page size. This, in my opinion, is more robust. Performance wise, we can add memoization later if it turns out to be a problem.

Reply:
💰

Reply:
@megsfulton do you see us having a need for a select all? Eg where we have say 299 unapproved comments and it selects all 299 of them, vs the first 20 on the page?

Reply:
@gwwar I'm not sure, but we can add it to our list of questions and keep an eye on the stats for bulk actions.

Reply:
we've discussed the possibility of the checkbox selecting the _current page_ + a message asking if the user want to select all, like gmail does. But indeed it's a question for power users. I think that for the rest of us, quick actions would be more useful.